src/gui_common/CustomCheckBox.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using Godot;
+
+/// <summary>
+///   A customized check box that changes icon when hovered / clicked.
+/// </summary>
+public class CustomCheckBox : Button
+{
+    private Texture unpressedNormal;
+    private Texture unpressedHovered;
+    private Texture unpressedClicked;
+    private Texture pressedNormal;
+    private Texture pressedHovered;
+    private Texture pressedClicked;
+
+    private bool pressing;
+    private CheckState lastCheckState;
+    private CheckState currentCheckState;
+
+    private enum CheckState
+    {
+        UnpressedNormal,
+        UnpressedHovered,
+        UnpressedClicked,
+        PressedNormal,
+        PressedHovered,
+        PressedClicked,
+        Disabled,
+    }
+
+    public override void _Ready()
+    {
+        unpressedNormal = GetIcon("UnpressedNormal", "CheckBox");
+        unpressedHovered = GetIcon("UnpressedHovered", "CheckBox");
+        unpressedClicked = GetIcon("UnpressedClicked", "CheckBox");
+        pressedNormal = GetIcon("PressedNormal", "CheckBox");
+        pressedHovered = GetIcon("PressedHovered", "CheckBox");
+        pressedClicked = GetIcon("PressedClicked", "CheckBox");
```

Comment:
There is the default `checked` and `unchecked` icon, perhaps those can be reused for `pressedNormal` and `unpressedNormal`.

Reply:
I prefer the current one that all custom icons are named consistently.